src/components/VideoItem.vue

```diff
@@ -177,12 +184,48 @@ export default {
             const request = objectStore.get(this.video.url.substr(-11));
             request.onsuccess = event => {
                 const video = event.target.result;
-                if (video && (video.currentTime ?? 0) > video.duration * 0.9) {
+                if (
+                    video &&
+                    (!this.getPreferenceBoolean("watchHistory", false) ||
```

Comment:
This is redundant if you set the watched till as the duration of the video, which you do above.

Reply:
Thank you!